tests/helpers.py

```diff
@@ -982,3 +995,99 @@ def delete_deploymentconfig(pod_obj):
+
+    """
+    with ThreadPoolExecutor(32) as p:
+        for obj in obj_list:
+            p.submit(obj.delete)
+    return True
+
+
+def memory_leak_analysis(median_dict):
+    """
+    Memory leak analysis after the execution of test case
+    Memory leak is analyzed based on "RES" value, i.e. list[7] in code
+    Args:
+         median_dict (dict): dict of worker nodes and respective median value
+    """
+    # dict to store memory leak difference for each worker
+    diff = {}
+    for worker in get_worker_nodes():
+        memory_leak_data = []
+        if os.path.exists(f"/tmp/{worker}-top-output.txt"):
+            with open(f"/tmp/{worker}-top-output.txt", "r") as f:
+                data = f.readline()
+                list = data.split(" ")
+                list = [i for i in list if i]
+                memory_leak_data.append(list[7])
+        else:
+            logging.info(f"worker {worker} memory leak file not found")
+            raise UnexpectedBehaviour
+        number_of_lines = len(memory_leak_data) - 1
+        # Get the start value form median_dict arg for respective worker
+        start_value = median_dict[f"{worker}"]
+        end_value = memory_leak_data[number_of_lines]
+        logging.info(f"Median value {start_value}")
+        logging.info(f"End value {end_value}")
+        # Convert the values to kb for calculations
+        if start_value.__contains__('g'):
+            start_value = float(1024 * 1024 * float(start_value[:-1]))
+        elif start_value.__contains__('m'):
+            start_value = float(1024 * float(start_value[:-1]))
+        else:
+            start_value = float(start_value)
+        if end_value.__contains__('g'):
+            end_value = float(1024 * 1024 * float(end_value[:-1]))
+        elif end_value.__contains__('m'):
+            end_value = float(1024 * float(end_value[:-1]))
+        else:
+            end_value = float(end_value)
+        # Calculate the percentage of diff between start and end value
+        # Based on value decide TC pass or fail
+        diff[worker] = ((end_value - start_value) / start_value) * 100
+        logging.info(f"Percentage diff in start and end value {diff[worker]}")
+        if diff[worker] <= 20:
+            logging.info(f"No memory leak in worker {worker} passing the test")
+        else:
+            logging.info(f"There is a memory leak in worker {worker}")
+            logging.info(f"Memory median value start of the test {start_value}")
+            logging.info(f"Memory value end of the test {end_value}")
+            raise UnexpectedBehaviour
+
+
+def memory_leak_median():
+    """
+    Calculate memory leak Median value by collecting the data for 120 sec
+    and find the median value which will be considered as starting point
+    to evaluate memory leak using "RES" value i.e. list[7] in code
+
+    Returns:
+        median_dict (dict): dict of worker nodes and respective median value
+    """
+    median_dict = {}
```

Comment:
please add logs and comments in the code

Reply:
observing a failure in ResourceLeftoversException because of parametrize, mentioned in previous comments as well.

Reply:
Problem resolved